cloudevents/formats/cbor-format.md

```diff
@@ -0,0 +1,199 @@
+risk of translation errors or information loss when the original format is not
+preserved.
+
+An extension specification that defines a secondary mapping rule for CBOR, and
+any revision of such a specification, MUST also define explicit mapping rules
+for all other event formats that are part of the CloudEvents core at the time of
+the submission or revision.
+
+If required, the CloudEvents type can be determined by inference using the rules
+from the mapping table. The value is compatible with the respective CloudEvents type when the
+mapping rules are fulfilled.
+
+### 2.4. CDDL Schema Validation
+
+The CloudEvents [CDDL][cddl-spec] for the spec is defined under
+[cloudevents.cddl](cloudevents.cddl) and contains the definitions for validating events in
+CBOR.
+
+## 3. Envelope
+
+Each CloudEvents event can be wholly represented as a CBOR data item.
+
+Such a representation MUST use the media type `application/cloudevents+cbor`.
+
+All REQUIRED and all not omitted OPTIONAL attributes in the given event MUST
+become members of the CBOR data item, with the respective CBOR item member name
+matching the attribute name, and the member's type and value being mapped using
+the [type system mapping](#22-type-system-mapping).
+
+OPTIONAL not omitted attributes MAY be represented as a `null` CBOR simple value.
+
+### 3.1. Handling of "data"
+
+The CBOR representation of the event "data" payload is determined by the runtime
+type of the `data` content and the value of the [`datacontenttype`
+attribute][datacontenttype].
+
+#### 3.1.1. Payload Encoding
+
+Before taking action, a [CBOR encoder][cbor-encoder] MUST first determine the runtime data
+type of the `data` content.
+
+If the implementation determines that the type of data is `Binary`, the value
+MUST be represented as a [Major type 2][cbor-major-types] value. If present, the `datacontenttype` MUST
+reflect the format of the original binary data.
+
+If the type of data is not `Binary`, the implementation will next determine
+whether the value of the `datacontenttype` attribute declares the `data` to
+contain CBOR-formatted content. Such a content type is defined as one having a
+[media subtype][rfc2045-sec5] equal to `cbor` or ending with a `+cbor` format
+extension. That is, a `datacontenttype` declares CBOR-formatted content if its
+media type, when stripped of parameters, has the form `*/cbor` or `*/*+cbor`.
+If the `datacontenttype` is unspecified, processing SHOULD proceed as if the
+`datacontenttype` had been specified explicitly as `application/cbor`.
+
+If the `datacontenttype` declares the data to contain CBOR-formatted content, a
+[CBOR encoder][cbor-encoder] MUST translate the data value to a [CBOR date item][cbor-data-item], and
+use the member name `data` to store it inside the CBOR representation. The data
+value MUST be stored directly as a CBOR data item, rather than as an encoded CBOR
+buffer (tagged or un-tagged). An implementation MAY fail to encode the
```

Comment:
MAY or MUST?
When wouldn't it fail if it can't encode it?

Reply:
This is how it is in the JSON format